protocol-designer/src/components/modals/CreateFileWizard/utils.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  getModuleType,
+  HEATERSHAKER_MODULE_TYPE,
+  TEMPERATURE_MODULE_TYPE,
+  THERMOCYCLER_MODULE_TYPE,
+} from '@opentrons/shared-data'
+import { OUTER_SLOTS_FLEX } from '../../../modules'
+import { isModuleWithCollisionIssue } from '../../modules'
+import {
+  FLEX_SUPPORTED_MODULE_MODELS,
+  DEFAULT_SLOT_MAP,
+} from './ModulesAndOtherTile'
+
+import type { ModuleType } from '@opentrons/shared-data'
+import type { FormModulesByType } from '../../../step-forms'
+import type { FormState } from './types'
+
+export const FLEX_TRASH_DEFAULT_SLOT = 'A3'
+
+export const getLastCheckedEquipment = (values: FormState): string | null => {
+  const hasTrash = values.additionalEquipment.includes('trashBin')
+
+  if (!hasTrash) {
+    return null
+  }
+
+  if (
+    values.modulesByType.heaterShakerModuleType.onDeck &&
+    values.modulesByType.thermocyclerModuleType.onDeck
+  ) {
+    return TEMPERATURE_MODULE_TYPE
+  }
+
+  if (
+    values.modulesByType.heaterShakerModuleType.onDeck &&
+    values.modulesByType.temperatureModuleType.onDeck
+  ) {
+    return THERMOCYCLER_MODULE_TYPE
+  }
+
+  if (
+    values.modulesByType.thermocyclerModuleType.onDeck &&
+    values.modulesByType.temperatureModuleType.onDeck
+  ) {
+    return HEATERSHAKER_MODULE_TYPE
+  }
+
+  return null
+}
+
+export const getCrashableModuleSelected = (
+  modules: FormModulesByType,
+  moduleType: ModuleType
+): boolean => {
+  const formModule = modules[moduleType]
+  const crashableModuleOnDeck =
+    formModule?.onDeck && formModule?.model != null
+      ? isModuleWithCollisionIssue(formModule.model)
+      : false
+
+  return crashableModuleOnDeck
+}
+
+export const getTrashBinOptionDisabled = (values: FormState): boolean => {
+  const allStagingAreasInUse =
+    values.additionalEquipment.filter(equipment =>
+      equipment.includes('stagingArea')
+    ).length === 4
```

Comment:
nit

```suggestion
    ).length === 4
```

maybe using const? 
```ts
const STAGINH_AREA_FULL = 4
```